web/src/api.ts

```diff
@@ -33,3 +35,34 @@ export const getAllListings = async (): Promise<Listing[]> => {
   const res = await fetch(`${process.env.REACT_APP_SERVER_URL}/listings`);
   return res.json();
 };
+
+/**
+ * Stores new merchant into the database.
+ */
+export const addMerchant = async (
+  merchant: MerchantPayload,
+  idToken: string
+): Promise<number> => {
+  const requestOptions = {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${idToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(merchant),
+  };
+  const res = await fetch(
+    `${process.env.REACT_APP_SERVER_URL}/merchants`,
+    requestOptions
+  );
+
+  if (res.status !== 201) {
+    throw new Error(Errors.SERVER_ERROR);
+  }
+
+  const location = res.headers.get('Location');
+  const merchantId = location?.substring(location?.lastIndexOf('/') + 1);
```

Comment:
should we add an extra field to res to return merchant id?
extracting from location is not very intuitive

Reply:
Yep! Actually that sounds good. I will make another PR for that?